File: ABC/ABC261/B.py
n = int(input())
s = [input() for _ in range(n)]
ok = True
for i in range(n):
    for j in range(i+1, n):
        if s[i][j] == 'W' and s[j][i] != 'L': ok = False
        elif s[i][j] == 'L' and s[j][i] != 'W': ok = False
        elif s[i][j] == 'D' and s[j][i] != 'D': ok = False

        if not ok:
            print("incorrect")
            exit(0)

print("correct")

File: ABC/ABC261/B.cpp
#include<iostream>
using namespace std;

#define rep(i, n) for(int i = 0; i < n; ++i)

void solve() {
	int n; cin >> n;
	char a[n][n];
	rep(i, n)rep(j, n)cin >> a[i][j];
	bool flag = false;
	rep(i, n) {
		for(int j = i + 1; j < n; ++j) {
			if(a[i][j]=='D' && a[j][i] != 'D') flag=true;
			if(a[i][j]=='W' && a[j][i]!='L') flag=true;
			if(a[i][j]=='L' && a[j][i]!='W') flag=true;			
		}
	}
	if(flag) cout << "incorrect" << endl;
	else cout << "correct" << endl;
}

int main(void){
  solve();
  return 0;
}
